worldedit-core/src/main/java/com/sk89q/worldedit/command/ClipboardCommands.java

```diff
@@ -479,7 +481,14 @@ public void place(
                     .apply(region.getMaximumPoint().subtract(region.getMinimumPoint()).toVector3())
                     .toBlockPoint());
             if (removeEntities) {
-                editSession.getEntities(new CuboidRegion(realTo, max)).forEach(Entity::remove);
+                // Collect entities on the current (command) thread, in case it benefits from asynchronous retrieval (unlikely)
+                final Collection<? extends Entity> entities = editSession.getEntities(new CuboidRegion(realTo, max));
+                // BukkitEntity#remove is synchronized, but it makes more sense to synchronize here beforehand in case many
+                // entities are affected. BukkitEntity will not synchronize if it's already called on the main thread.
+                TaskManager.taskManager().sync(() -> {
+                    entities.forEach(Entity::remove);
+                    return true;
+                });
```

Comment:
Not super happy about this, don't we have a better method to remove entities in a region? The paste method below also has the same problem.

Reply:
Valid, I guess I can create one on the Extent or EditSession